nautobot/dcim/factory.py

```diff
@@ -0,0 +1,101 @@
+import factory
+from factory.django import DjangoModelFactory
+
+from nautobot.dcim.models import Location, LocationType, Region, Site
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance, UniqueFaker
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+
+
+class LocationTypeFactory(DjangoModelFactory):
+    class Meta:
+        model = LocationType
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    name = UniqueFaker(
+        "random_element",
+        elements=(
+            "Continent",
+            "Country",
+            "State",
+            "City",
+            "Building",
+            "Floor",
+            "Room",
+            "RackGroup",
+            "Rack",
+            "BabyRack",
+        ),
+    )
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_parent", random_instance(LocationType), None)
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
+
+
+class LocationFactory(DjangoModelFactory):
+    class Meta:
+        model = Location
+        exclude = (
+            "has_parent",
+            "has_site",
+            "has_tenant",
+            "has_description",
+        )
+
+    name = UniqueFaker(
+        "random_element",
+        elements=(
+            "Africa",
+            "North America",
+            "Asia",
+            "Australia",
+            "Europe",
+            "United States",
+            "Russia",
+            "China",
+            "Building 1",
+            "Building 2",
+            "Building 3",
+            "Building 4",
+            "Building 5",
+            "Building 6",
+            "Floor 1",
+            "Floor 2",
+            "Floor 3",
+            "Floor 4",
+            "Floor 5",
+        ),
+    )
+    location_type = random_instance(LocationType)
```

Comment:
To create valid instances, might need a `sequence` here as well, since (to continue on the suggested Building/Floor/Room LocationType hierarchy) it's not valid to create a Floor or Room without a parent, so we'd need to create at least one Building first, etc.

Additionally, my preference would be to have the location names more accurately mapped to the location-type, so that we don't end up with a Room called "Building 1" or a Building called "Africa".